pkg/activator/handler/handler.go

```diff
@@ -169,54 +165,45 @@ func (a *activationHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		Host:   host,
 	}
 
-	_, ttSpan := trace.StartSpan(r.Context(), "throttler_try")
-	ttStart := time.Now()
-
-	tryContext := r.Context()
+	tryContext, trySpan := trace.StartSpan(r.Context(), "throttler_try")
 	if a.endpointTimeout > 0 {
 		var cancel context.CancelFunc
-		tryContext, cancel = context.WithTimeout(r.Context(), a.endpointTimeout)
+		tryContext, cancel = context.WithTimeout(tryContext, a.endpointTimeout)
 		defer cancel()
 	}
+
+	tryStart := time.Now()
 	err = a.throttler.Try(tryContext, revID, func() {
-		var (
-			httpStatus int
-		)
+		trySpan.End()
+		a.logger.Debugf("Waiting for throttler took %v time", time.Since(tryStart))
 
-		ttSpan.End()
-		a.logger.Debugf("Waiting for throttler took %v time", time.Since(ttStart))
+		probeCtx, probeSpan := trace.StartSpan(r.Context(), "probe")
```

Comment:
IMO I think its not a big diff either way, but: https://github.com/knative/serving/pull/2726#discussion_r271934313 :)

Reply:
yeah, I like the enders be deferred, since if we panic in startspan, it won't be closed... but at that point we probably don't care much for the span as well.